src/lex/replace.rs

```
@@ -166,56 +166,104 @@ pub fn replace(
+                                    .iter()
+                                    .cloned()
+                                    .map(|t| Ok(location.with(t))),
+                            );
+                            new_pending.append(&mut pending);
+                            pending = new_pending;
+                            continue;
+                        }
+                        // TODO: so many allocations :(
+                        Some(Definition::Function { .. }) => {
+                            ids_seen.insert(id);
+                            let func_replacements = replace_function(
+                                definitions,
+                                id,
+                                location,
+                                &mut pending,
+                                &mut inner,
+                            );
+                            let mut func_replacements: VecDeque<_> =
+                                func_replacements.into_iter().collect();
+                            func_replacements.append(&mut pending);
+                            pending = func_replacements;
+                            continue;
+                        }
+                        None => {}
                     }
-                    None => {}
                 }
             }
+            Ok(Locatable {
+                data: Token::HashHash(true),
+                ..
+            }) => {
+                let preceding_tok = loop {
+                    match replacements.pop() {
+                        Some(Ok(Locatable {
+                            data: Token::Whitespace(_),
+                            ..
+                        })) => continue,
+                        Some(Ok(Locatable { data: token, .. })) => break token,
+                        None | Some(Err(_)) => {
+                            return wrap_error(&location, CppError::HashHashMissingParameter(true))
+                        }
```

Hmm, in the `Some(Err)` case I think we should return the original error instead of swallowing it.